modules/renter/repair.go

```diff
@@ -150,6 +151,21 @@ func (f *file) chunkHosts(chunk uint64) []modules.NetAddress {
 	return old
 }
 
+// expiringContracts returns the contracts that will expire soon.
+// TODO: what if contract has fully expired?
+func (f *file) expiringContracts(height types.BlockHeight) []fileContract {
+	f.mu.RLock()
+	defer f.mu.RUnlock()
+
+	var expiring []fileContract
+	for _, fc := range f.contracts {
+		if height > fc.WindowStart-renewThreshold {
```

Comment:
does it also make sense to check if the host is online (by looking at a table in the hostdb, not by doing network I/O on-the-fly), and to check whether the contract has fully expired? I'm guessing the logic here is too simple.